subsys/dfu/dfu_target/src/dfu_target_mcuboot.c

```diff
@@ -185,13 +189,31 @@ int dfu_target_mcuboot_done(bool successful)
 	}
 
 	if (successful) {
+#ifdef CONFIG_FLASH_HAS_EXPLICIT_ERASE
+		stream_flash_erase_flatten_t erase_function = stream_flash_erase_page;
+#else
+		stream_flash_erase_flatten_t erase_function = NULL;
+#endif
+#ifdef CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE
```

Comment:
Don't do that. Make subsys/dfu/dfu_target/Kconfig option that will be set depending on these and use it in code.
The general idea was that this is not directly used in code but rather propagates to configuration of modules, unfortunately due to time and stuff I had to cut corners and cut on the Kconfig options added.
Only Kconfig options defined directly for Module should affect it in code.

Reply:
And define the function that does all of that or nothing and just call it.

Reply:
You can't do it from Kconfig, if you have internal RRAM on a device and external SPI flash, then you need to query the device because one needs erase and the other needs flatten

Reply:
If your intention is to remove data, rather then erase as device requires it, the flatten has the logic inside, so you can just call it
https://github.com/zephyrproject-rtos/zephyr/blob/417a9e81eca20b85a6f26e417156815cfe2c932a/drivers/flash/flash_util.c#L64-L81

Reply:
@de-nordic Usage of these Kconfig allow use to keep only what is needed in compiled code.